packages/utils/telemetry-utils/src/utils.ts

```diff
@@ -30,3 +31,39 @@ export function logIfFalse(
 	logger.send(newEvent);
 	return false;
 }
+
+/**
+ * sampletext
+ */
+export function createSampledLoggerSend(
+	logger: ITelemetryBaseLogger,
+	shouldSampleEventCallback: () => boolean,
+) {
```

Comment:
Debating with myself if this would feel more ergonomic if it returned an `IBaseTelemetryLogger` so the returned object had a `send()` method instead of just being a function to call. Interested in thoughts from others.

Reply:
Updated to do as your suggested instead.